astropy/io/fits/hdu/hdulist.py

```diff
@@ -293,7 +320,7 @@ def __contains__(self, item):
 
         try:
             self._try_while_unread_hdus(self.index_of, item)
-        except KeyError:
+        except (KeyError, TypeError):
```

Comment:
I think it would be fine, instead of changing this, to change `index_of` back to just raising a `KeyError` if the type of key is invalid.  I'm not entirely sure what motivated me to change that in the first place.

Thanks for fixing this!